src/main/scala/com/gu/elasticsearchmonitor/ClusterHealth.scala

```diff
@@ -22,15 +22,35 @@ object ClusterHealth {
       .url(s"$host/_cluster/health")
       .build()
 
+    val shardAllocationRequest = new Request.Builder()
+      .url(s"$host/_cluster/allocation/explain")
+      .build()
+
+    val nodeInfoRequest = new Request.Builder()
+      .url(s"$host/_nodes")
+      .build()
+
     val clusterHealthResponse = Try(httpClient.newCall(clusterHealthRequest).execute())
 
     val result = clusterHealthResponse match {
       case Success(response) if response.code == 200 =>
         val root = mapper.readTree(response.body.string)
         logger.info("Fetched the cluster health")
+        val clusterStatus = root.get("status").asText
+        if (clusterStatus != "green") {
+          val requestAttempts = for {
+            shardRequest <- Try(httpClient.newCall(shardAllocationRequest).execute())
+            nodeRequest <- Try(httpClient.newCall(nodeInfoRequest).execute())
+          } yield {
+            logger.warn(s"Cluster is in $clusterStatus. Shard allocation info:\n${shardRequest.body.string}\nNode info:\n${nodeRequest.body.string}")
+          }
+          if (requestAttempts.isFailure) {
+            logger.error(s"Failed to obtain debug information about cluster status due to ${requestAttempts.failed.get}")
```

Comment:
I think we can avoid this by using `requestAttempts.recover` instead of `if (requestAttempts.isFailure) {`

Reply:
Yep, that's much neater, thanks for the suggestion 👍  https://github.com/guardian/elastic-search-monitor/pull/6/commits/de9363e3256e3c96adc9706472bdde7b1a5be311